modules/store/src/store.ts

```diff
@@ -246,6 +268,27 @@ export function select<T, Props, K>(
       );
     }
 
-    return mapped$.pipe(distinctUntilChanged());
+    return mapped$.pipe(
+      distinctUntilChanged((previous, current) => {
+        if (
+          current instanceof SelectedValues &&
+          previous instanceof SelectedValues
```

Comment:
I needed to use a distinct class to hold multiple selected values so that they could be distinguished from a single selected array value.